modules/host/host.go

```diff
@@ -120,8 +310,74 @@ type Host struct {
 	resourceLock sync.RWMutex
 }
 
-// New returns an initialized Host.
-func New(cs modules.ConsensusSet, tpool modules.TransactionPool, wallet modules.Wallet, address string, persistDir string) (*Host, error) {
+// establishDefaults configures the default settings for the host, overwriting
+// any existing settings.
+func (h *Host) establishDefaults() error {
+	// Configure the settings object.
+	h.settings = modules.HostSettings{
+		MaxDuration: defaultMaxDuration,
+		WindowSize:  defaultWindowSize,
+
+		Collateral:             defaultCollateral,
+		ContractPrice:          defaultContractPrice,
+		DownloadBandwidthPrice: defaultDownloadBandwidthPrice,
+		UploadBandwidthPrice:   defaultUploadBandwidthPrice,
+	}
+
+	// Generate signing key, for revising contracts.
+	sk, pk, err := crypto.GenerateKeyPair()
+	if err != nil {
+		return err
+	}
+	h.secretKey = sk
+	h.publicKey = types.SiaPublicKey{
+		Algorithm: types.SignatureEd25519,
+		Key:       pk[:],
+	}
+	_, err = rand.Read(h.sectorSalt[:])
+	if err != nil {
+		return err
+	}
+
+	// Subscribe to the consensus set.
+	return h.initConsensusSubscription()
+}
+
+// initDB will check that the database has been initialized and if not, will
+// initialize the database.
+func (h *Host) initDB() error {
+	return h.db.Update(func(tx *bolt.Tx) error {
+		// Return nil if the database is already initialized. The database can
+		// be safely assumed to be initialized if the storage obligation bucket
+		// exists.
+		bso := tx.Bucket(bucketStorageObligations)
```

Comment:
It would work in this case, though it would stop working if you added extra code like inserting some data or something (like the genesis block). I think when I created this function I had an intention of adding some base data, but later changed the architecture so it wasn't needed.